samza-core/src/main/java/org/apache/samza/storage/blobstore/BlobStoreRestoreManager.java

```diff
@@ -254,8 +263,10 @@ static CompletableFuture<Void> restoreStores(String jobName, String jobId, TaskN
         throw new SamzaException(String.format("Error deleting store directory: %s", storeDir), e);
       }
 
+      // If getDeletedBlobs is enabled - always restore so that we get all the blobs, including the deleted blobs,
+      // immediately restore it locally and backup to create new checkpoint.
```

Comment:
Minor: "backup to create .." is not happening here, clarify comment.

Reply:
Thanks, clarified.